resources/certificate/certificate.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificate
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/injection/clients/dynamicclient"
+
+	"knative.dev/reconciler-test/pkg/feature"
+)
+
+var (
+	certificateGVR = schema.GroupVersionResource{
+		Group:    "cert-manager.io",
+		Version:  "v1",
+		Resource: "certificates",
+	}
+)
+
+type RotateCertificate struct {
+	Certificate types.NamespacedName
+}
```

Comment:
```suggestion
type RotateCertificate types.NamespacedName
```
nit. so we could avoid the nesting.
```
foo := certificate.RotateCertificate{
	Certificate: types.NamespacedName{
		Namespace: "foo",
		Name:      "bar",
	},
}
```

But not blocking...

Reply:
Like things like "...Ctx" in https://github.com/knative-sandbox/reconciler-test/pull/548 could have been avoided with proper object/struct to work with instead of direct types

Reply:
ack.
/unhold